lib/msf/ui/debug.rb

```diff
@@ -145,7 +148,44 @@ def self.datastore(framework, driver)
           content
         )
       rescue StandardError => e
-        section_build_error('Failed to extract Datastore', e)
+        build_section(
+          'Module/Datastore',
+          ERROR_BLURB,
+          section_build_error('Failed to extract Datastore', e)
+        )
+      end
+
+      def self.database_stats(framework)
+        if framework.db&.active
+          output = "Total Workspaces: #{::Mdm::Workspace.count}\n"
+          output += "Total Hosts: #{::Mdm::Host.count}\n"
+          output += "Total Vulnerabilities: #{::Mdm::Vuln.count}\n"
+          output += "Total Notes: #{::Mdm::Note.count}\n"
+          output += "Total Services: #{::Mdm::Service.count}\n\n"
+
+          output += "Workspaces:\n\n"
+          ::Mdm::Workspace.all.each do |workspace|
```

Comment:
If the reporter has hundreds of workspaces, this will be too much information. The original approach was to intentionally only grab 10 or so workspaces as an indication database usage.

We should probably prioritize the _current_ workspace, and a few other workspaces. This should also be an ordered list by their ID to ensure determinism when generating reports

Reply:
Sounds good to me!
Taking the workspaces from `framework.db.workspaces` or `::Mdm::Workspace.take(10).each` results in a list that is already sorted by workspace ID 👍

Reply:
By default rails and postgres won't guarantee order unless explicitly required:

```
>> ::Mdm::Workspace.all.to_sql
=> "SELECT \"workspaces\".* FROM \"workspaces\""
```

For the `take(...)` command

```
>> ActiveRecord::Base.logger = Logger.new STDOUT
=> 
#<Logger:0x00007f7f7141f378
>> ::Mdm::Workspace.take(10)
D, [2021-11-22T16:07:43.928394 #17389] DEBUG -- :   Mdm::Workspace Load (0.5ms)  SELECT "workspaces".* FROM "workspaces" LIMIT $1  [["LIMIT", 10]]
```

Note that both sql expressions are missing an `order by` clause, and if we cross-reference with the postgres documentation:

> If the ORDER BY clause is specified, the returned rows are sorted in the specified order. If ORDER BY is not given, the rows are returned in whatever order the system finds fastest to produce. (See ORDER BY Clause below.)
> https://www.postgresql.org/docs/9.5/sql-select.html

So we'll want an explicit order on the id